src/sidebar/components/ShareDialog/FileInput.tsx

```diff
@@ -0,0 +1,60 @@
+import {
+  FileGenericIcon,
+  IconButton,
+  Input,
+  InputGroup,
+} from '@hypothesis/frontend-shared';
+import type { MutableRef } from 'preact/hooks';
+import { useId } from 'preact/hooks';
+
+export type FileInputProps = {
+  onFileSelected: (file: File) => void;
+  fileInputRef: MutableRef<HTMLInputElement | null>;
+  disabled?: boolean;
+};
+
+export default function FileInput({
+  onFileSelected,
+  disabled,
+  fileInputRef,
+}: FileInputProps) {
+  const inputId = useId();
+
+  return (
+    <>
+      <input
+        ref={fileInputRef}
+        accept=".json"
+        type="file"
+        disabled={disabled}
+        className="invisible absolute w-0 h-0"
+        data-testid="file-input"
+        onChange={e => {
+          const files = (e.target as HTMLInputElement)!.files;
+          if (files !== null && files.length > 0) {
+            onFileSelected(files[0]);
+          }
+        }}
+      />
+      <label htmlFor={inputId}>Select Hypothesis export file:</label>
+      <InputGroup>
+        <Input
+          id={inputId}
+          onClick={() => fileInputRef.current?.click()}
+          readonly
+          disabled={disabled}
+          value={fileInputRef.current?.files?.[0]?.name ?? 'Select a file'}
```

Comment:
When no file is selected, the native control shows a representation of the _value_ of the field (eg. "No file selected") whereas the empty-state text here is an _instruction_ ("Select a file").